lib/api/objectPost.js

```diff
@@ -0,0 +1,137 @@
+    let parsedContentLength = 0;
+    const passThroughStream = new PassThrough();
+    const requestType = request.apiMethods || 'objectPost';
+    const valParams = {
+        authInfo,
+        bucketName,
+        objectKey: formData.key,
+        requestType,
+        request,
+    };
+    const canonicalID = authInfo.getCanonicalID();
+
+
+    log.trace('owner canonicalID to send to data', { canonicalID });
+    return standardMetadataValidateBucketAndObj(valParams, request.actionImplicitDenies, log,
+        (err, bucket, objMD) => {
+            const responseHeaders = collectCorsHeaders(headers.origin,
+                method, bucket);
+
+            if (err && !err.AccessDenied) {
+                log.trace('error processing request', {
+                    error: err,
+                    method: 'metadataValidateBucketAndObj',
+                });
+                return callback(err, responseHeaders);
+            }
+            if (bucket.hasDeletedFlag() && canonicalID !== bucket.getOwner()) {
+                log.trace('deleted flag on bucket and request ' +
+                    'from non-owner account');
+                return callback(errors.NoSuchBucket);
+            }
+
+            return async.waterfall([
+                function countPOSTFileSize(next) {
+                    if (!request.fileEventData || !request.fileEventData.file) {
+                        return next();
+                    }
+                    request.fileEventData.file.on('data', (chunk) => {
+                        parsedContentLength += chunk.length;
+                        passThroughStream.write(chunk);
+                    });
+
+                    request.fileEventData.file.on('end', () => {
+                        passThroughStream.end();
+                        // Setting the file in the request avoids the need to make changes to createAndStoreObject's
+                        // parameters and thus all it's subsequent calls. This is necessary as the stream used to create
+                        // the object is that of the request directly; something we must work around
+                        // to use the file data produced from the multipart form data.
+                        /* eslint-disable no-param-reassign */
+                        request.fileEventData.file = passThroughStream;
+                        /* eslint-disable no-param-reassign */
+                        // Here parsedContentLength will have the total size of the file
+                        // This is used when calculating the size of the object in createAndStoreObject
+                        request.parsedContentLength = parsedContentLength;
+                        return next();
+                    });
+                    return undefined;
+                },
+                function objectCreateAndStore(next) {
+                    writeContinue(request, request._response);
+                    return createAndStoreObject(request.bucketName,
```

Comment:
I see in this function many places hardcoded for put object. Maybe we should have another function or all the hardcoded places for put should be replaced with post.

I'm not sure if we want to differentiate the action

Reply:
I'm not sure either, but given it's usage in other APIs, I followed that trend.

Reply:
But this function sets most of the attributes from the headers and query string, that are not present for the POST.

Also it sets `metadataStoreParams.originOp = 's3:ObjectCreated:Put';` that ight need to change for Post if we want to differentiate them. 

We should find how this is used, it's probably a point to raise and to keep in mind

Reply:
Actions here?

Make a ticket to further explore how createAndStoreObject can be differentiated for the actions that call it?

Reply:
It will need to be changed later anyways to send request without Content-Length.

But we mostly need to investigate the usage of `metadataStoreParams.originOp = 's3:ObjectCreated:Put';`
to know if we need to differentiate put and post